src/NumericalAlgorithms/Interpolation/PredictedZeroCrossing.cpp

```diff
@@ -16,4 +17,31 @@ double predicted_zero_crossing_value(const std::vector<double>& x_values,
   return -coefficients[0]/coefficients[1];
 }
 
+DataVector predicted_zero_crossing_value(
+    const std::deque<double>& x_values,
+    const std::deque<DataVector>& y_values) {
+  ASSERT(x_values.size() == y_values[0].size(),
+         "The x_values and y_values must be of the same size");
```

Comment:
This is the wrong ASSERT, and is the same problem as your original version.

x_values.size() should be equal to y_values.size().  y_values[0].size() should have nothing to do with x_values.size().

Reply:
Yeah, that makes sense now, I should have an ASSERT that checks for the y_values and x_values to be the same size.